File: backend/app/schemas/theme.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel


class ThemeBase(BaseModel):
    category: str
    summary: str

class ThemeCreate(ThemeBase):
    audience_id: int

class ThemeUpdate(BaseModel):
    category: Optional[str] = None
    summary: Optional[str] = None

class ThemeResponse(ThemeBase):
    id: int
    audience_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class ThemeWithPosts(ThemeResponse):
    posts: List[dict]  # List of posts with relevance scores 

File: frontend/src/types/theme.ts
export interface Theme {
  id: number;
  category: string;
  summary: string;
  created_at: string;
  updated_at: string;
  audience_id: number;
  post_count?: number;
  posts?: {
    id: number;
    title: string;
    url: string;
    score: number;
    num_comments: number;
    created_utc: string;
  }[];
} 